### PR TITLE
feat: add project scan capability, preserve workspace scan function [ROAD-1239]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

Previously, the Snyk Test action in the context menu triggered a workspace scan. This had some unwanted side-effects for customers:

- diagnostics of all projects were cleared during the scan
- resources / scan time for the whole workspace were occupied

Now, Snyk Test only scans the selected project, and a new menu item is added for workspace scan (see screenshot)

Must be merged/released AFTER https://github.com/snyk/snyk-ls/pull/203

### Checklist

- [x] Linted
- [x] CHANGELOG.md updated

### Screenshots / GIFs

image

### CODE REVIEW

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java [hunk 1]
```diff
@@ -58,10 +59,9 @@ public void triggerScan(IWorkbenchWindow window) {
       runSnykWizard();
     } else {
       try {
-        executeCommand("snyk.workspace.scan", new ArrayList<>());
-
         if (window == null) {
           window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+          executeCommand("snyk.workspace.scan", new ArrayList<>());
         }
         if (window == null) {
```
Comment (asaf92) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:66]: Why 2 conditions?

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java [hunk 2]
```diff
@@ -118,12 +121,8 @@ public void addTrustedPaths(SnykTrustedFoldersParams param) {
     var trustedPaths = storedTrustedPaths.split(File.pathSeparator);
     var pathSet = new HashSet<>(Arrays.asList(trustedPaths));
     pathSet.addAll(Arrays.asList(param.getTrustedFolders()));
-    Preferences.getInstance().store(Preferences.TRUSTED_FOLDERS,
-        pathSet.stream()
-        .filter(s -> !s.isBlank())
-        .map(s -> s.trim())
-        .distinct()
-        .collect(Collectors.joining(File.pathSeparator)));
+    Preferences.getInstance().store(Preferences.TRUSTED_FOLDERS, pathSet.stream().filter(s -> !s.isBlank())
```
Comment (asaf92) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:124]: Personally I prefer the copped version but don't worry about it

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java [hunk 3]
```diff
@@ -70,15 +70,18 @@ public void triggerScan(IWorkbenchWindow window) {
...
+        IProject project = null;
+        if (firstElement instanceof JavaProject) {
+          project = ((JavaProject) firstElement).getProject();
+        }
+        
         if (firstElement instanceof IProject) {
-          IProject project = (IProject) firstElement;
-          runForProject(project.getName());
+          project = (IProject) firstElement;
         }
```
Comment (asaf92) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:80]: If `firstElement` is a `JavaProject`, both conditions are true and the first assignment gets overwritten.

Comment (asaf92) [reply plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:80]: Maybe you wanted an `else if`?

**Review by asaf92 — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java (hunk 1 above)
Comment (bastiandoetsch) [reply plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:66]: Hm... probably wrong.

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java (hunk 3 above)
Comment (bastiandoetsch) [reply plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:80]: How? A JavaProject is not in the same type hierarchy as IProject.

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java (hunk 2 above)
Comment (bastiandoetsch) [reply plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:124]: me, too, but the auto-formatter didn't ask me :(

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java (hunk 3 above)
Comment (asaf92) [reply plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:78]: You're right, I was confused by `IJavaProject` and the similar `getProject()` function

**Review by asaf92 — APPROVED** [review]

**Pull request merged**